src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -1398,21 +1400,51 @@ private void checkStudyContainsNode(UUID studyUuid, UUID nodeUuid) {
         }
     }
 
-    public List<Pair<UUID, String>> getReportUuidsAndNames(UUID nodeUuid, boolean nodeOnlyReport) {
-        return networkModificationTreeService.getReportUuidsAndNames(nodeUuid, nodeOnlyReport);
+    @Transactional(readOnly = true)
+    public List<ReporterModel> getNodeReport(UUID nodeUuid, boolean nodeOnlyReport) {
+        return getSubReportersByNode(nodeUuid, nodeOnlyReport);
     }
 
-    public List<ReporterModel> getNodeReport(UUID nodeUuid, boolean nodeOnlyReport) {
-        List<Pair<UUID, String>> reportUuidsAndNames = getReportUuidsAndNames(nodeUuid, nodeOnlyReport);
-        return reportUuidsAndNames.stream().map(reportInfo -> {
-            ReporterModel reporter = reportService.getReport(reportInfo.getLeft(), reportInfo.getRight());
-            ReporterModel newReporter = new ReporterModel(reporter.getTaskKey(), reportInfo.getRight(), reporter.getTaskValues());
-            reporter.getReports().forEach(newReporter::report);
-            reporter.getSubReporters().forEach(newReporter::addSubReporter);
-            return newReporter;
+    private List<ReporterModel> getSubReportersByNode(UUID nodeUuid, boolean nodeOnlyReport) {
+        List<ReporterModel> subReporters = getSubReportersByNode(nodeUuid);
+        if (subReporters.isEmpty()) {
+            return subReporters;
+        } else if (nodeOnlyReport) {
+            return List.of(subReporters.get(subReporters.size() - 1));
+        } else {
+            if (subReporters.get(0).getTaskKey().equals(ROOT_NODE_NAME)) {
+                return subReporters;
+            }
+            Optional<UUID> parentUuid =  networkModificationTreeService.getParentNodeUuid(UUID.fromString(subReporters.get(0).getTaskKey()));
+            return parentUuid.isEmpty() ? subReporters : Stream.concat(getSubReportersByNode(parentUuid.get(), false).stream(), subReporters.stream()).collect(Collectors.toList());
+        }
+    }
+
+    private List<ReporterModel> getSubReportersByNode(UUID nodeUuid) {
+        AbstractNode nodeInfos = networkModificationTreeService.getNode(nodeUuid);
+        ReporterModel reporter = reportService.getReport(nodeInfos.getReportUuid(), nodeInfos.getId().toString());
+        Map<String, Pair<List<ReporterModel>, List<ReporterModel>>> subReportersByNode = new LinkedHashMap<>();
+        reporter.getSubReporters().forEach(subReporter -> subReportersByNode.put(getNodeIdFromReportKey(subReporter), Pair.of(new ArrayList<>(), new ArrayList<>())));
+        reporter.getSubReporters().forEach(subReporter -> {
+            if (subReporter.getTaskKey().endsWith(NETWORK_MODIFICATION_TYPE_REPORT)) {
+                subReportersByNode.get(getNodeIdFromReportKey(subReporter)).getLeft().addAll(subReporter.getSubReporters());
+            } else {
+                subReportersByNode.get(getNodeIdFromReportKey(subReporter)).getRight().addAll(subReporter.getSubReporters());
+            }
```

Comment:
So, all network modifications report will be displayed first, and then all computations report after ?

Reply:
Fixed